processor/src/test/java/com/linecorp/decaton/processor/runtime/ProcessingContextImplTest.java

```diff
@@ -113,31 +145,60 @@ private static void safeAwait(CountDownLatch latch) {
 
     @Test(timeout = 5000)
     public void testPush_Level1_Sync() throws InterruptedException {
-        ProcessingContextImpl<HelloTask> context = context((ctx, task) -> { /* noop */ });
-
-        CompletableFuture<Void> future = context.push(TASK);
-        assertTrue(future.isDone());
+        TraceHandle handle = new TestTraceHandle("testPush_Level1_Sync");
+        try {
+            ProcessingContextImpl<HelloTask> context = context(handle,
+                                                               new NamedProcessor("Noop", (ctx, task) ->
+                                                                       assertEquals("testPush_Level1_Sync-Noop",
```

Comment:
I think calling asserts inside such closure/callback that is expected to be executed as part of testing target's logic isn't a good pattern because:

* the execution of the closure itself is a part of target's logic and we it doesn't calls it , assertion isn't executed at all
* if the target try-catch the execution and crush the raised exception, assertion won't work properly (not very sure though)

so it's better to have a AtomicReference or something and capture the value inside, test its value in the test method's body.

btw, I think what we want to test for `ProcessingContextImplTest` is that it properly interacts with `TraceHandle` instance, which means beyond the work calling its method like `processStart` is out of scope (it's a responsibility of the driver implementation) of this test case, so this check isn't necessary I think?

Reply:
We do need to check that the processor is actually executed by the same thread that called the `TraceHandle` methods one way or another, as that's an important property and fairly easy to break. But agreed with just capturing the value here and then asserting afterwards; will do that.